libpkg/pkg_add.c

```diff
@@ -56,14 +59,137 @@
 #define NOCHANGESFLAGS	(UF_IMMUTABLE | UF_APPEND | SF_IMMUTABLE | SF_APPEND)
 #endif
 
+static merge_status
+merge_with_external_tool(const char *merge_tool, struct pkg_config_file *lcf,
+    size_t lcf_len, struct pkg_config_file *rcf, char *localconf, char **mergedconf)
+{
+	pid_t wait_res;
+	int status;
+	FILE *inout[2];
+
+	char *tmpdir = getenv("TMPDIR");
+	if (tmpdir == NULL)
+		tmpdir = "/tmp";
+
+	int output_fd;
+	char output_path[MAXPATHLEN];
+	off_t output_sz;
+
+	strlcpy(output_path, tmpdir, sizeof(output_path));
+	strlcat(output_path, "/OUTPUT.XXXXXXXXXX", sizeof(output_path));
+	output_fd = mkstemp(output_path);
```

Comment:
all the path should be relative to rootfd otherwise it will not work with pkg -r see the other code creating temp files

Reply:
It feels like I should use `mkostempsat`? But it is never used anywhere in pkg.